### PR TITLE
Hods 91 : Added organisations endpoints. 

### PR DESCRIPTION
**Pull request by parkeradam**

### CODE REVIEW

**Comment by owen-opencast** [comment]
Looks good to me, it'd be good if Andy could take a look though

**Review by owen-opencast — APPROVED** [review]

File: resources/public/api/conf/1.0/application.raml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #%RAML 1.0
 ---
 
-title: Individuals Integration Framework Test Support
+title: Integration Framework Test Support
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:4]: Great 👍 . Keeps Lisa happy :-)

File: resources/public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -294,15 +294,261 @@ uses:
...
+          type: string
+          required: true
+          example: Bearer <BEARER_TOKEN>
+        Content-Type:
+          description: "Specifies the content-type of the body."
+          type: string
+          required: true
+          example: application/json
+      queryParameters:
+        startYear:
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:339]: We need to remove the start and end years as they are not applicable for SSP. They only exist in the IF schema for suture use by other consumers. 

The scenario we are interested in from the IF schema usage description is 

- Neither startYear or endYear: will return the data for the most recent 2 years (2, 1, or 0 years can be returned) before the current SA tax year (CY-1)

File: resources/public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -294,15 +294,261 @@ uses:
...
+
+  /organisations/self-assessment/{utr}/return/details:
+    uriParameters:
+      utr:
+        description: "A Unique Tax Reference"
+        type: string
+        required: true
+        example: "1234567890"
+    post:
+      displayName: "Create a self-assessment organisations payload"
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:324]: suggest `Create a self-assessment return details payload`? 

More applicable to the endpoint description?

File: resources/public/api/conf/1.0/application.raml [hunk 4]
```diff
@@ -294,15 +294,261 @@ uses:
...
+          type: string
+          required: true
+          example: Bearer <BEARER_TOKEN>
+        Content-Type:
+          description: "Specifies the content-type of the body."
+          type: string
+          required: true
+          example: application/json
+      queryParameters:
+        startYear:
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:393]: No start year and end year required here.

File: resources/public/api/conf/1.0/application.raml [hunk 5]
```diff
@@ -294,15 +294,261 @@ uses:
...
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:447]: No start and end year required here.

File: resources/public/api/conf/1.0/application.raml [hunk 6]
```diff
@@ -294,15 +294,261 @@ uses:
...
+          type: integer
+          required: true
+          example: 2019
+      body:
+        application/json:
+          schema: !include schemas/organisations/create-corporation-tax-detail.json
+          example: !include examples/organisations/create-corporation-tax-detail.json
+      responses:
+        201:
+          description: The created corporation-tex details
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:463]: tex? 

Perhaps rename to `corporation-tax  return details`

File: resources/public/api/conf/1.0/application.raml [hunk 7]
```diff
@@ -294,15 +294,261 @@ uses:
...
+          type: string
+          required: true
+          example: application/json
+      body:
+        application/json:
+          schema: !include schemas/organisations/create-corporation-tax-company-detail.json
+          example: !include examples/organisations/create-corporation-tax-company-detail.json
+      responses:
+        201:
+          description: The created corporation-tex details
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:505]: tex? 

Perhaps rename to `The created corporation-tax company details`?

File: resources/public/api/conf/1.0/application.raml [hunk 8]
```diff
@@ -294,15 +294,261 @@ uses:
...
+                    code: INVALID_REQUEST
+  /organisations/employers/employee/counts:
+    uriParameters:
+      utr:
+        description: "A Unique Tax Reference"
+        type: string
+        required: true
+        example: "1234567890"
+    post:
+      displayName: "Create a employee count payload"
```
Comment (ajchilts) [inline resources/public/api/conf/1.0/application.raml:527]: replace `a` with `an` 

Replace `count` with `counts`

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: resources/public/api/conf/1.0/application.raml (hunk 6 above)
Comment (parkeradam) [reply resources/public/api/conf/1.0/application.raml:463]: Good typo spot xD

**Review by ajchilts — APPROVED** [review]

**Pull request merged**